### PR TITLE
feat: Add option enforce_security_group_inbound_rules_on_private_link_traffic

### PR DESCRIPTION
**Pull request by ricardoleal**
## Description
Add `enforce_security_group_inbound_rules_on_private_link_traffic` to `aws_lb`

## Motivation and Context
When using Security Group for network load balancers there is a need to configure whether the inbound rules are enforced for PrivateLink traffic.

https://github.com/hashicorp/terraform-provider-aws/issues/33766
https://github.com/hashicorp/terraform-provider-aws/pull/33767

## Breaking Changes
No.

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects
- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: versions.tf [hunk 1]
```diff
@@ -4,7 +4,7 @@ terraform {
   required_providers {
     aws = {
       source  = "hashicorp/aws"
-      version = ">= 5.23"
+      version = ">= 5.30"
```
Comment (bryantbiggs) [inline versions.tf:7]: we will wait until [v5.31.0](https://github.com/hashicorp/terraform-provider-aws/pull/34140) is released before adding support

File: variables.tf [hunk 2]
```diff
@@ -104,6 +104,12 @@ variable "load_balancer_type" {
   default     = "application"
 }
 
+variable "enforce_security_group_inbound_rules_on_private_link_traffic" {
+  description = "Indicates whether inbound security group rules are enforced for traffic originating from a PrivateLink. Only valid for Load Balancers of type network. The possible values are on and off."
+  type        = string
+  default     = "off"
```
Comment (bryantbiggs) [inline variables.tf:110]: This is an option value so we should default to `null`
```suggestion
  default     = null
```

**Review by bryantbiggs — CHANGES_REQUESTED** [review]
A few updates and we'll need to wait for the next release of the provider

File: versions.tf (hunk 1 above)
Comment (ricardoleal) [reply versions.tf:7]: Ok, let's wait

Comment (ricardoleal) [reply versions.tf:7]: @bryantbiggs 
https://github.com/hashicorp/terraform-provider-aws/releases/tag/v5.31.0 is out.
do we keep `version = ">= 5.30"` of we should bump to `version = ">= 5.31"` ?

Comment (bryantbiggs) [reply versions.tf:7]: 5.31 please

Comment (ricardoleal) [reply versions.tf]: All done !

**Review by bryantbiggs — APPROVED** [review]
thank you!

**Comment by antonbabenko** [comment]
This PR is included in version 9.3.0 :tada:

**Pull request merged**